### DISCUSSION TITLE
Add config to name game folder with game name

### DISCUSSION THREAD

**Discussion by samuelT2** [pyscsd/steamCloudSaveDownloader#7] [Ideas]
Hey. What a great util, thank you :)

Would be nice if one could save the saves with the game name in the folder name, eg. ".data/730_Counter-Strike_2/..."

- Configurable with a config param (Default: Behaviour as is, On: Add game name)
- Replace spaces with "_"
- Limit folder name to X chars

**Comment by hhhhhojeihsu** [comment]
Hi samuel

Thanks for your feedback but I'm not sure what's your request. My understanding is you want 4 additional functionalities.

- Only download file for a game that matches specific patterns(regex)
- I have no idea what this is
  > Configurable with a config param (Default: Behaviour as is, On: Add game name) 
- Replace all occurrence of specific character within filename and replace with other character (e.g. `Counter Strike 2` to `Counter_Strike_2`)
- Limit the length of folder name (e.g. `SomeLongAssFolderName` to `SomeLong`)

I agree with the first one, but could not think of practical usage for the remaining ones. Could you provide a few ideas of how it should workout or your usage scenario?

**Reply by samuelT2** [reply]
Sorry for the confusion and thank you for replying.

No, it's a single feature request - let me try to explain it better, the use case is as follows: Currently the downloader creates a folder structure inside data, eg:

```
.data/7
.data/220
.data/400
.data/550
.data/730
```

Now, if I want to restore a save from a specific game, I first have to look up the id of that game - let's say I'm looking for the CS2 configs, I need to google the id first (which is 730).
It would be nice, if the structure would look like this inside data to avoid that extra step:

```
.data/7_game_name
.data/220_game_name
.data/400_game_name
.data/550_game_name
.data/730_Counter_Strike_2
```

To not mess up the existing functionality, the extra game tag naming could be configurable via config. If not set, the naming stays as is (default for this bool is false).

```
[General]
save_dir = ./data
append_game_name = true
...
```

To avoid running into OS specific path length issues, the append_game_name would also cut any string longer than x (not sure what that value should be).

Would be a nice to have for me, nothing more. If it's too much of work, no problem (or I can work on a pull request later).

**Reply by hhhhhojeihsu** [reply]
Hi samuel,

Thanks for the detailed explanation. I agree that it is tedious to lookup appID for the game. Let's go with your suggestion. I'll convert this discussion into an issue to track it.
